src/contexts/wallet.tsx

```diff
@@ -129,9 +131,20 @@ export const WalletProvider = ({ children = null as any }) => {
   async function sign(xdr: string): Promise<string> {
     if (connected) {
       setTxStatus(TxStatus.SIGNING);
-      let result = await signTransaction(xdr, { networkPassphrase: network.passphrase });
-      setTxStatus(TxStatus.SUBMITTING);
-      return result;
+      try {
+        let result = await signTransaction(xdr, { networkPassphrase: network.passphrase });
+        setTxStatus(TxStatus.SUBMITTING);
+        return result;
+      } catch (e: any) {
+        if (e == 'User declined access') {
```

Comment:
nitpick : ideally use `===`

Reply:
will fix in followup PR so we can merge this